src/program.rs

```diff
@@ -34,6 +34,26 @@ use std::ffi::OsString;
 use std::io::{self, Read};
 use std::result::Result;
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ColorOpt {
+    Auto,
+    Always,
+    Never,
+}
+
+impl std::str::FromStr for ColorOpt {
+    type Err = &'static str;
```

Comment:
I believe the only constraint that structopts places on this `Err` type is that it needs to implement `Display`. We _could_ introduce indirection here by returning a `ColorOptParseError::UnsupportedColor` and give that a `Display` impl that just returns the string which actually gets printed by structops, but I'm not sure it's really worth it. 

This entire `FromStr` implementation will go away if/when, as Viktor suggests, we move to a recent `clap` version, as it can derive parsing (& appropriate error messages) directly from the enum.

Reply:
Do as you think is best :slightly_smiling_face: it was mostly a remark